nesc/whip6/platforms/parts/mcu/cc26xx/timers/HalEventCountPrv.nc

```diff
@@ -170,12 +192,20 @@ implementation {
         return SUCCESS;
     }
 
-    event void Timer.fired() {
-        call Timer.startWithTimeoutFromLastTrigger(pollingInterval);
-        if (running & TIMER_A)
-            readValue(TIMER_A);
-        if (running & TIMER_B)
-            readValue(TIMER_B);
+    event async void ChannelAInterrupt.triggered() {
+        atomic {
```

Comment:
Interrupt handlers in NesC should by default be atomic. Check if this is not the case for our platform. I do not know whether clearing the flag is even correct.

Reply:
They're not, interrupts can preempt other interrupts. But there are no other interrupts that would create a data race with this interrupt. I feel like leaving the atomic keyword just in case.